lib/reverse_sentence.rb

```diff
@@ -1,4 +1,52 @@
-# A method to reverse the words in a sentence, in place.
+# A method to reverse each word in a sentence, in place.
+def reverse_words(my_words)
+  return my_words unless my_words
+
+  position = 0
+  string_length = my_words.length
+
+  while position < my_words.length
+    until my_words[position] != ' ' || position >= string_length
+      position += 1
+    end
+
+    start_point = position
+
+    until my_words[position] == ' ' || position >= string_length
+      position += 1
+    end
+
+  end_point = position - 1
+
+  reverse_word(my_words, start_point, end_point)
+  end
+
+  return
+end
+
+def reverse_word(my_words, start_point, end_point)
```

Comment:
This method uses three temporary storage values: two to hold the index values i and j and one to hold the temporary character while swapping. So, regardless of how many characters are in the input or what characters are in the string, this method will always need 3 temporary storage values. Hence, the space complexity is O(1) or constant.

To deduce the time complexity, we count the number of operations in this function as compared to either the input value or input size. In this case, it would depend on the number of characters from start_point to end_point If there are n characters between start_point and end_point then the while loop will run n/2 times or in other words the method will swap characters n/2 times. That's why, the time complexity will be O(n/2) or by dropping the constant, O(n) or linear.